src/utilities/indexer/writeQuery.ts

```diff
@@ -0,0 +1,36 @@
+import { QUERY_SIZE } from './queryFromIndexer';
+
+export function writeQuery({
+  entity,
+  alias,
+  fields,
+  querySize = QUERY_SIZE,
+  offset = 0,
+  filter,
+  fragments,
+}: {
+  entity: string;
+  alias?: string;
+  fields: string[];
+  filter?: string;
+  querySize?: number;
+  offset?: number;
+  fragments?: string[];
+}) {
+  // BlockIDs are strings, this means that "42" > "1000"
+  // Sadly, time_stamps can only be use to order queries of blocks.
+  // Specially for queries with many matches, it is important to request an order of the results, to avoid duplicates and assure totality.
+  // It does not matter what the order criterion is as long as it is consistent.
+  // ID_ASC exists for all entities, so it should work as a general approach.
+  return `
+  query {
+    ${alias ? alias + ':' : ''} ${entity}(orderBy: ID_ASC, first: ${querySize}, offset: ${offset}, ${filter ? 'filter: ' + filter : ''}) {
+      totalCount
+      nodes {
+        ${fields.join('\n        ')}
+      }
+    }
+  }
+  ${fragments ? fragments.join('\n') : ''}
```

Comment:
I think we only have two distinct queries, and to "unify" them we had to introduce three optional fields, for the small upside of not repeating "orderBy, totalCount, nodes". Does it make sense to keep this function instead of just explicitly writing two queries? Yet another upside would be the possibility to try out the query directly, without the need to log the return value of this function.

Reply:
I was trying to make a generic function that will work on the other apps. 

Also for the "pagination" (=increasing the offset) a function like this seems to me better than search for "offset: " on a string and replace the value.

Reply:
It was like you suggest until this point: https://github.com/galaniprojects/ctypehub/pull/743/commits/253d674e8693e156ff013eac1947087f0cd939e2

Reply:
> I was trying to make a generic function that will work on the other apps.

This is a classical case of good intentions which led to overengineering, unfortunately. Now this is definitely overengineered.

The pagination issue could be simply addressed by creating the query already with the right offset

```js
const query = `... offset: ${offset} ...`

Reply:
But the `query` should come from outside of the `src/utilities/indexer/queryFromIndexer.ts` file.  

I thought of writing some weird string as the value of the offset and then search it and replace it for pagination.

Reply:
[I’ve pushed my proposal on a different branch](https://github.com/galaniprojects/ctypehub/commit/99805476d7664699559066e0b9eab3112df89aa6). I believe it is immediately clear to anyone knowing GraphQL, and there’s no code duplication.

Reply:
Well, there is definitely code duplication of the explicit query. But it is alright. 

I think both versions are ok.

Reply:
Feel free to push this changes to my branch.

Reply:
I just cherry picked it.